From e4508f2106aae42de102607018d1f25285dd13dc Mon Sep 17 00:00:00 2001
From: cheney <11722200@qq.com>
Date: Thu, 20 Mar 2025 14:22:25 +0800
Subject: [PATCH 257/534] Change: Rework the previous DTS changes related to
 WiFi/BT by merging all relevant modifications into
 eic7702-deepcomputing-fml13v03.dts, while reverting the changes in
 eic7x-die0-pinctrl.dtsi and eic7x-die1-pinctrl.dtsi.

---
 .../eswin/eic7702-deepcomputing-fml13v03.dts  | 45 +++++++++++++++++++
 .../boot/dts/eswin/eic7x-die0-pinctrl.dtsi    |  5 ---
 .../boot/dts/eswin/eic7x-die1-pinctrl.dtsi    |  6 +--
 3 files changed, 48 insertions(+), 8 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts b/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts
index 89ea3cf635fd..1ebbfeff875b 100644
--- a/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts
+++ b/arch/riscv/boot/dts/eswin/eic7702-deepcomputing-fml13v03.dts
@@ -491,6 +491,51 @@ &d1_lpcpu {
 	status = "okay";
 };
 
+&pinctrl_die0_gpio27_default {
+	mux {
+		groups = "gpio27_group";
+		function = "gpio27_func";
+	};
+	conf {
+		groups = "gpio27_group";
+		drive-strength = <2>;
+		output-high;
+		input-enable = <0>;
+		bias-pull-up = <1>;
+		bias-pull-down = <0>;
+	};
+};
+
+&pinctrl_die1_gpio16_default {
+	mux {
+		groups = "gpio16_group";
+		function = "gpio16_func";
+	};
+	conf {
+		groups = "gpio16_group";
+		drive-strength = <2>;
+		output-high;
+		input-enable = <0>;
+		bias-pull-up = <1>;
+		bias-pull-down = <0>;
+	};
+};
+
+&pinctrl_die1_gpio18_default {
+	mux {
+		groups = "gpio18_group";
+		function = "gpio18_func";
+	};
+	conf {
+		groups = "gpio18_group";
+		drive-strength = <2>;
+		output-high;
+		input-enable = <0>;
+		bias-pull-up = <1>;
+		bias-pull-down = <0>;
+	};
+};
+
 &pcie {
 	/* m.2 ssd key m */
 	status = "okay";
diff --git a/arch/riscv/boot/dts/eswin/eic7x-die0-pinctrl.dtsi b/arch/riscv/boot/dts/eswin/eic7x-die0-pinctrl.dtsi
index 9e8e0ce060a1..8409a0886b7d 100644
--- a/arch/riscv/boot/dts/eswin/eic7x-die0-pinctrl.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7x-die0-pinctrl.dtsi
@@ -799,11 +799,6 @@ mux {
 					groups = "gpio27_group";
 					function = "gpio27_func";
 				};
-				conf {
-					groups = "gpio27_group";
-					input-enable = <0>;
-					bias-pull-up = <1>;
-				};
 			};
             pinctrl_die0_gpio28_default: gpio28-default{
 				mux {
diff --git a/arch/riscv/boot/dts/eswin/eic7x-die1-pinctrl.dtsi b/arch/riscv/boot/dts/eswin/eic7x-die1-pinctrl.dtsi
index 8a1c568c104e..fcb3d855c105 100644
--- a/arch/riscv/boot/dts/eswin/eic7x-die1-pinctrl.dtsi
+++ b/arch/riscv/boot/dts/eswin/eic7x-die1-pinctrl.dtsi
@@ -717,7 +717,7 @@ mux {
 				conf {
 					groups = "gpio16_group";
 					input-enable = <0>;
-					bias-pull-up = <1>;
+					bias-pull-down = <1>;
 				};
 			};
             pinctrl_die1_gpio18_default: gpio18-default{
@@ -727,8 +727,8 @@ mux {
 				};
 				conf {
 					groups = "gpio18_group";
-					input-enable = <0>;
-					bias-pull-up = <1>;
+					input-enable = <1>;
+					bias-pull-down = <1>;
 				};
 			};
             pinctrl_die1_gpio19_default: gpio19-default{
-- 
2.53.0

